tests/performance/PerformanceTest.cpp

```diff
@@ -0,0 +1,179 @@
+#include <olp/dataservice/read/DataRequest.h>
+#include <olp/dataservice/read/PartitionsRequest.h>
+
+#include <mocks/KeyValueCache.h>
+
+using namespace olp;
+
+struct CatalogClientTestConfiguration {
+  int number_of_requests;
+  int calling_thread_count;
+  int task_scheduler_capacity;
+  bool use_no_cache;
+};
+
+std::ostream& operator<<(std::ostream& stream,
+                         const CatalogClientTestConfiguration& config) {
+  return stream << "CatalogClientTestConfiguration("
+                << ".calling_thread_count=" << config.calling_thread_count
+                << ", .task_scheduler_capacity="
+                << config.task_scheduler_capacity
+                << ", .number_of_requests=" << config.number_of_requests
+                << ", .use_no_cache=" << config.use_no_cache << ")";
+}
+
+namespace {
+const std::string kKeyId("");
+const std::string kKeySecret("");
+const std::string kLayerId("");
+const std::string kPartitionId("");
+}  // namespace
+
+class CatalogClientTest
+    : public ::testing::TestWithParam<CatalogClientTestConfiguration> {
+ public:
+  static void SetUpTestSuite() {
+    s_network =
+        client::OlpClientSettingsFactory::CreateDefaultNetworkRequestHandler();
+  }
+  static void TearDownTestSuite() {
+    OLP_SDK_LOG_INFO_F("CatalogClientTest", "Network use count is %ld",
+                       s_network.use_count());
+    s_network = nullptr;
+  }
+
+  http::NetworkProxySettings GetLocalhostProxySettings() {
+    return olp::http::NetworkProxySettings()
+        .WithHostname("http://localhost:3000/http_proxy")
+        .WithUsername("test_user")
+        .WithPassword("test_password")
+        .WithType(olp::http::NetworkProxySettings::Type::HTTP);
+  }
+
+  std::shared_ptr<dataservice::read::CatalogClient> CreateCatalogClient() {
+    client::HRN hrn("fake");
+
+    auto parameter = GetParam();
+
+    std::shared_ptr<olp::thread::TaskScheduler> task_scheduler =
+        olp::client::OlpClientSettingsFactory::CreateDefaultTaskScheduler(
+            parameter.task_scheduler_capacity);
+
+    olp::authentication::Settings settings;
+    settings.task_scheduler = task_scheduler;
+    settings.network_request_handler = s_network;
+
+    olp::client::AuthenticationSettings auth_settings;
+    auth_settings.provider = olp::authentication::TokenProviderDefault(
+        kKeyId, kKeySecret, std::move(settings));
+
+    auto client_settings = std::make_shared<olp::client::OlpClientSettings>();
+    client_settings->authentication_settings = auth_settings;
+    client_settings->task_scheduler = task_scheduler;
+    client_settings->network_request_handler = s_network;
+    client_settings->proxy_settings = GetLocalhostProxySettings();
+
+    std::shared_ptr<olp::cache::KeyValueCache> cache = CreateCache();
+
+    return std::make_shared<dataservice::read::CatalogClient>(
+        hrn, client_settings, cache);
+  }
+
+  std::shared_ptr<olp::cache::KeyValueCache> CreateCache() {
```

Comment:
This is definitely not a real behavior testing strategy.
Mocked elements in a performance test is a very new experience for me, never seen this before.
Our key concept in the SDK is that we have cache to speed things up and basically everyone is using it, so why do you want to remove it?